### PR TITLE
bpo-33336: Legalize MOVE command

### PR DESCRIPTION
**Pull request by mcepl**
Just checking whether this is a good idea at all. If so, I will add metadata required.

https://bugs.python.org/issue33336

### CODE REVIEW

**Comment by mcepl** [comment]
I would love to eventually have some way how to make moving messages universally possible (https://bugs.python.org/issue33327), but this is the least what I can hope for.

**Review by vstinner — COMMENTED** [review]
You should add a ".. versionchanged:: 3.7" markup near https://docs.python.org/dev/library/imaplib.html#imaplib.IMAP4.uid (Doc/library/imaplib.rst) to document that MOVE is now supported.

**Comment by mcepl** [comment]
> You should add a ".. versionchanged:: 3.7" markup near https://docs.python.org/dev/library/imaplib.html#imaplib.IMAP4.uid (Doc/library/imaplib.rst) to document that MOVE is now supported.

I still hold that this is not change in API. We never documented which IMAP commands are supported, so there is no change in documentation.

I would like this bugfix be ported to other older versions of Python, which won't happen for change of API.

And yes, if calling this change of API makes you stop stalling merging of this PR (I thought we went through all this on IRC, but apparently that discussion didn't change anything), I am willing to shut up and let it be limited to 3.8. Better that than nothing.

File: Misc/NEWS.d/next/Library/2018-04-27-22-18-38.bpo-33336.T8rxn0.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+``imaplib`` now allows ``MOVE`` command in ``IMAP4.uid()`` and potentially
+as a name of supported method of ``IMAP4`` object.
```
Comment (vstinner) [inline Misc/NEWS.d/next/Library/2018-04-27-22-18-38.bpo-33336.T8rxn0.rst]: Would you mind to add a reference to the RFC 6851? Example:

``imaplib`` now allows ``MOVE`` command in ``IMAP4.uid()`` (RFC 6851: IMAP MOVE Extension)
and potentially as a name of supported method of ``IMAP4`` object.

**Review by vstinner — APPROVED** [review]
LGTM.

I don't know the IMAP protocol nor the imaplib module. I would be more confortable if someone from the "email team", @warsaw or someone else review this PR. But this PR was created in April, 3 monhs ago, and didn't get any review whereas the PR looks safe and good (to me).

I suggest to backport the change to 2.7, 3.6 and 3.7 branches.

I will merge this PR next week except if someone show up against it.

**Comment by vstinner** [comment]
@bitdancer, @warsaw: Would you mind to review this imaplib PR?

**Pull request merged**